autoload/go/guru.vim

```diff
@@ -217,11 +213,19 @@ endfunc
 
 " run_guru runs the given guru argument
 function! s:run_guru(args) abort
+
+  let old_gopath = $GOPATH
+  let $GOPATH = go#path#Detect()
   if go#util#has_job()
-    return s:async_guru(a:args)
+    let abort = s:async_guru(a:args)
```

Comment:
I was returning with the previous one, as we might need it in the future and it's handy. You can also just return however I'm not sure how to unset it easily (as Vim doesn't have defer like we have in Go)